src/uwtools/drivers/mpas_init.py

```diff
@@ -0,0 +1,178 @@
+        yield [
+            symlink(target=Path(target), linkname=self._rundir / linkname)
+            for linkname, target in self._driver_config.get("files_to_link", {}).items()
+        ]
+
+    @task
+    def namelist_file(self):
+        """
+        The namelist file.
+        """
+        fn = "namelist.init_atmosphere"
+        yield self._taskname(fn)
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        yield None
+        stop_time = self._cycle + timedelta(
+            hours=self._driver_config["boundary_conditions"]["length"]
+        )
+        d = {
+            "nhyd_model": {
+                "config_start_time": self._cycle.strftime("%Y-%m-%d_%H:00:00"),
+                "config_stop_time": stop_time.strftime("%Y-%m-%d_%H:00:00"),
+            }
+        }
+        namelist = self._driver_config.get("namelist", {})
+        values = YAMLConfig(d)
+        values.update_values(namelist.get("update_values", {}))
+        namelist["update_values"] = values.data
+        self._create_user_updated_config(
+            config_class=NMLConfig,
+            config_values=namelist,
+            path=path,
+        )
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.boundary_files(),
+            self.files_copied(),
+            self.files_linked(),
+            self.namelist_file(),
+            self.runscript(),
+            self.streams_file(),
+        ]
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        self._write_runscript(path=path, envvars={})
+
+    @task
+    def streams_file(self):
+        """
+        The streams file.
+        """
+        fn = "streams.init_atmosphere"
+        yield self._taskname(fn)
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        yield file(path=Path(self._driver_config["streams"]["path"]))
+        render(
+            input_file=Path(self._driver_config["streams"]["path"]),
+            output_file=path,
+            values_src=self._driver_config["streams"]["values"],
+        )
+
+    # Private helper methods
+
+    @property
+    def _driver_name(self) -> str:
+        """
+        Returns the name of this driver.
+        """
+        return STR.mpasinit
+
+    @property
+    def _resources(self) -> Dict[str, Any]:
+        """
+        Returns configuration data for the runscript.
+        """
+        return {
+            "account": self._config["platform"]["account"],
+            "rundir": self._rundir,
+            "scheduler": self._config["platform"]["scheduler"],
+            **self._driver_config.get("execution", {}).get("batchargs", {}),
+        }
```

Comment:
General question, I don't see a `_resources` private helper method implemented for any other driver anymore. I think it's just in the base class driver. I wasn't sure if this was an abstract method we needed to implement for each specific driver. I have it implemented in my jedi driver currently as well.

@maddenp-noaa

Reply:
I factored out `_resources` to the drive base class with some light refactoring when the `chgres_cube` driver was added. It's no longer an abstract class, but a concrete one in the base class.